### PR TITLE
[SUP-673] Update styles on upload page

### PR DESCRIPTION
**Pull request by nawatts**
A few minor style changes to the upload page (https://app.terra.bio/#upload).

* Vertically align icons with text in section headings
* Made the spacing between section heading icon and text consistent with the spacing between text and "Create new ..." button
* Remove extra padding at the bottom of collection cards
* Size the "Create new collection" button based on its text

## Before
Screen Shot 2022-06-13 at 10 01 17 AM

## After
Screen Shot 2022-06-13 at 10 01 05 AM

### CODE REVIEW

File: src/pages/Upload.js [hunk 1]
```diff
@@ -386,16 +387,12 @@ const CollectionSelectorPanel = _.flow(
             style: _.merge(styles.workspaceTile, prefix === selectedCollection ? styles.workspaceTileSelected : {}),
             onClick: () => setCollection(prefix),
             variant: prefix === selectedCollection ? 'light' : 'dark'
-          }, [
-            h4({
-              style: { margin: '0 0 1rem 0' }
-            }, [prefix])
-          ])
+          }, [prefix])
```
Comment (nawatts) [inline src/pages/Upload.js:385]: `h4` didn't seem appropriate here since this wasn't serving as a heading.

**Review by kyuksel — APPROVED** [review]
Looks great! Verified that the changes did not introduce any regression as reported by `axe DevTools`. There are however existing contrast issues classified as `serious`. 

image

In the same vein, another pre-existing issue (at least to my eyes) is the vertical alignment of `Change` on the right and `WORKSPACE` on the left below where they aren't vertically centered like the workspace name text.

image


I'll leave it up to you whether to fix them opportunistically or not.

**Comment by nawatts** [comment]
Styled those buttons like the cards found on other pages to resolve the contrast issues. Also got rid of the nested interactive elements (the "Change" links) that axe warned about.

Screen Shot 2022-06-14 at 11 06 16 AM

**Review by kyuksel — APPROVED** [review]
🤩 ❤️

**Review by MatthewBemis — APPROVED** [review]

**Pull request merged**